### DISCUSSION TITLE
Could you provide a demo to implement the interface of AfterEachIteration(Void)?

### DISCUSSION THREAD

**Discussion by haochengd** [SuperElastix/elastix#842] [Q&A]
In some cases, we need to output some information in each iteration through the interface. Although in the manual there is some description in 7.4, however, it is not clear. Is that possible to elaborate on some examples of how to do that?

**Comment by ntatsisk** [comment]
Hi @haochengd, `AfterEachIteration(Void)` is meant to be used by developers that want to implement new filters to be used internally by Elastix. So, it does not regard to the public interface of Elastix.

If the information that you are referring to is, for example, the value of the metric for each iteration, then you might more helpful to set the parameter `(WriteTransformParametersEachIteration "true")`. You can find more information one that in the following older discussions:

- https://groups.google.com/g/elastix-imageregistration/c/7LW-LAGveas/m/TuF8NiiQDwAJ
- https://groups.google.com/g/elastix-imageregistration/c/WU4r7r6emv0/m/llPL9WsgBAAJ

Let us know if that helps.
